tests/test_pgsql_parse.py

```diff
@@ -0,0 +1,616 @@
+    def test_pgsql_parse_update(self):
+        self.assertEqual(
+            parse_and_gen("UPDATE my_table SET the_value = DEFAULT"),
+            "UPDATE my_table SET the_value = DEFAULT",
+        )
+        self.assertEqual(
+            parse_and_gen(
+                "UPDATE tictactoe SET board[1:3][1:3] = '{{,,},{,,},{,,}}' "
+                "WHERE game = 1"
+            ),
+            "UPDATE tictactoe SET board[1:3][1:3] = '{{,,},{,,},{,,}}' "
+            "WHERE game = 1",
+        )
+        self.assertEqual(
+            parse_and_gen(
+                "UPDATE accounts SET "
+                "(contact_first_name, contact_last_name) = "
+                "(SELECT first_name, last_name "
+                "FROM salesmen WHERE salesmen.id = accounts.sales_id)"
+            ),
+            "UPDATE accounts SET "
+            "(contact_first_name, contact_last_name) = "
+            "(SELECT first_name, last_name "
+            "FROM salesmen WHERE salesmen.id = accounts.sales_id)",
+        )
+        self.assertEqual(
+            parse_and_gen("UPDATE my_table SET id = 5; DELETE FROM my_table"),
+            "UPDATE my_table SET id = 5; DELETE FROM my_table",
+        )
+        self.assertEqual(
+            parse_and_gen(
+                "UPDATE dataset SET a = 5 WHERE id IN "
+                "(SELECT * from table_one) OR age IN (select * from table_two)"
+            ),
+            "UPDATE dataset SET a = 5 WHERE id IN "
+            "(SELECT * from table_one) OR age IN (select * from table_two)",
+        )
+        self.assertEqual(
+            parse_and_gen("UPDATE dataset SET a = 5 FROM extra WHERE b = c"),
+            "UPDATE dataset SET a = 5 FROM extra WHERE b = c",
+        )
+        self.assertEqual(
+            parse_and_gen(
+                "UPDATE users SET one_thing = $1, second_thing = $2 "
+                "WHERE users.id = ?"
+            ),
+            "UPDATE users SET one_thing = $1, second_thing = $2 "
+            "WHERE users.id = ?",
+        )
+        self.assertEqual(
+            parse_and_gen(
+                "UPDATE users SET something_else = $1 WHERE users.id = ?"
+            ),
+            "UPDATE users SET something_else = $1 WHERE users.id = ?",
+        )
+        self.assertEqual(
+            parse_and_gen(
+                "UPDATE users SET something_else = "
+                "(SELECT a FROM x WHERE uid = users.id LIMIT 1) "
+                "WHERE users.id = ?"
+            ),
+            "UPDATE users SET something_else = "
+            "(SELECT a FROM x WHERE uid = users.id LIMIT 1) "
+            "WHERE users.id = ?",
+        )
+        self.assertEqual(
+            parse_and_gen("UPDATE x SET a = 1, b = 2, c = 3"),
+            "UPDATE x SET a = 1, b = 2, c = 3",
+        )
+        self.assertEqual(
+            parse_and_gen("UPDATE x SET z = now()"), "UPDATE x SET z = now()"
+        )
+
+    @unittest.skip("unsupported SQL constructs")
```

Comment:
What's the intent here? To eventually fix some of them? You should use xfail for such.

As for parsing unsupported commands like `PREPARE a123 AS SELECT a` -- you shouldn't fail on the first one, you should test failing on *every one* of them.

Reply:
Intent is to save this collection of tests if we decide to support these features in the future.

Will change to xfail.